completions/_quick-lint-js

```diff
@@ -0,0 +1,58 @@
+#compdef _quick-lint-js quick-lint-js
+
+# Copyright (C) 2020  Matthew Glazar
+# See end of file for extended copyright information.
+
+# This script is used to provide command line completion for
+# quick-lint-js when using zsh shell
+
+typeset -A opt_args
+
+function _quick-lint-js() {
+        local curcontext="$curcontext" state line
+        local -i ret=1
+        local -a args
+
+        # don't autocomplete these if other arguments are present
+        local I1="-h --help -v --version"
+        local I2="--output-format"
+        local I3="--lsp --lsp-server"
+
+        args+=(
+        "(- *)"{-h,--help}"[Print help message]"
+        "(- *)"{-v,--version}"[Print version information]"
+        "($I1 $I2 --lsp --lsp-server)"{--lsp,--lsp-server}"[Run quick-lint-js in LSP server mode]"
+        "($I1 $I3)--exit-fail-on=[Fail with a non-zero exit code if any of these errors are found (default\: \"all\")]"
+        "($I1 $I3)--output-format=[Format to print feedback where FORMAT is one of\: gnu-like (default if omitted), vim-qflist-json]:format:->format"
+        "($I1)--vim-file-bufnr=[Select a vim buffer for outputting feedback]")
```

Comment:
The outcome of the exclusions seems a bit weird:

* `quick-lint-js --lsp --<TAB>` will suggest `--exit-fail-on` and `--vim-file-bufnr`. Neither of these options work with `--lsp`.
* `quick-lint-js --output-format=gnu-like --<TAB>` will suggest  `--exit-fail-on` and `--vim-file-bufnr`. `--help` and `--version` also make sense to suggest (in my opinion).

I think the following changes should make it complete how I expect:
```suggestion
        "(--exit-fail-on --lsp --lsp-server --vim-file-bufnr)"{--lsp,--lsp-server}"[Run quick-lint-js in LSP server mode]"
        "(--lsp --lsp-server)--exit-fail-on=[Fail with a non-zero exit code if any of these errors are found (default\: \"all\")]"
        "(--lsp --lsp-server --output-format)--output-format=[Format to print feedback where FORMAT is one of\: gnu-like (default if omitted), vim-qflist-json]:format:->format"
        "(--lsp --lsp-server)--vim-file-bufnr=[Select a vim buffer for outputting feedback]")
```

Reply:
`Implemented`